docs/glossary.md

```diff
@@ -0,0 +1,53 @@
+# Terminology
+
+### Jobs 
+
+Jobs, or Primary Jobs, are special types of tasks run in simulations.
+
+### Tasks
+
+Tasks are any thing run within simulations.
+
+### Taskflow
+
+Taskflows are a collection of tasks and jobs which are run for simulations.
+
```

Comment:
Taskflows are collections of tasks run to orchestrate a give simulation workflow.